packages/components/modal/src/use-modal.ts

```diff
@@ -129,6 +129,12 @@ export function useModal(originalProps: UseModalProps) {
     },
   });
 
+  useEffect(() => {
+    if (isOpen) {
+      onOpenChange?.(isOpen);
+    }
+  }, [isOpen]);
+
```

Comment:
I came across this interesting resource on the React docs ([https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)) that suggests alternative approaches to `useEffect` in some scenarios. It might be worth considering if it applies to this particular use case.

In the example provided, they achieve a similar outcome by storing information from previous renders.  I took a stab at adapting it here (untested):

```jsx
const [prevIsOpen, setPrevIsOpen] = useState(isOpen);
if (isOpen !== prevIsOpen) {
  setPrevIsOpen(isOpen);
  if (isOpen) {
    onOpenChange?.(isOpen);
  }
}
```

What do you think?

Reply:
Thanks, It seems a very nice idea to avoid useEffect. I'll test and see if it works.

Reply:
I tested, actually the onOpenChange is eventually changing the isOpen state. And we shouldn't directly update parent state inside child without useEffect.

I tried though and got the warning : "Cannot update a component (`App`) while rendering a different component (`NextUI.Modal`)".  

Let me know if I am missing something.